TheMovieDb/Models/MoviesList.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  MovieList.swift
+//  TheMovieDb
+//
+//  Created by Alex on 11/4/21.
+//
+
+import Foundation
+
+struct MovieItem: Decodable {
+    let id: String
+    let name: String
+
+    enum CodingKeys: String, CodingKey {
+        case id
+        case name
+        case url
+    }
+    
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        let url = try container.decode(String.self, forKey: .url)
+        guard let id = url.matches("(\\d+)/$").first?.last else {
+            throw NSError(domain: "", code: 0, userInfo: nil)
+        }
+        self.id = id
+        self.name = try container.decode(String.self, forKey: .name)
+    }
+}
+
+struct MoviesList: Codable {
+    let results : [Movie]?
+}
+
+extension String {
+    func matches(_ pattern: String) -> [[String]] {
```

Comment:
What is the purpose of this function?

Reply:
This and the next block are doing nothing, I left them before thinking that might gonna need it, and then I forgot to remove it.